src/arraydialog.cpp

```diff
@@ -90,3 +90,206 @@ void ArrayDialog::newRows(int num) {
+            if (item->column() > end_col)
+            {
+                end_col = item->column();
+            }
+        }
+
+        for (int row=start_row; row <= end_row; ++row)
+        {
+            for (int col=start_col; col <= end_col; ++col)
+            {
+                QTableWidgetItem *item = ui.tableWidget->item(row, col);
+                copied_text += item->text();
+                if (col < ui.tableWidget->columnCount() - 1)
+                {
+                    copied_text += "\t";
+                }
+            }
+            copied_text += "\n";
+        }
+        QClipboard *clipboard = QGuiApplication::clipboard();
+        clipboard->setText(copied_text);
+    }
+
+    // Cut
+    if ((event->modifiers() == Qt::CTRL) && (event->key() == Qt::Key_X))
+    {
+        // Check to ensure selection exists.
+        QItemSelectionModel *selection_model = ui.tableWidget->selectionModel();
+        if (selection_model)
+        {
+            QModelIndexList indexList = selection_model->selectedIndexes();
+            if (indexList.count() <= 0)
+            {
+                return;
+            }
+        }
+        else
+        {
+            return;
+        }
+
+        QString copied_text = "";
+        QList<QTableWidgetItem *> list = ui.tableWidget->selectedItems();
+
+        int start_row = list[0]->row();
+        int end_row = list[0]->row();
+        int start_col = list[0]->column();
+        int end_col = list[0]->column();
+        for (QTableWidgetItem *item : list)
+        {
+            if (item->row() < start_row)
+            {
+                start_row = item->row();
+            }
+            if (item->row() > end_row)
+            {
+                end_row = item->row();
+            }
+            if (item->column() < start_col)
+            {
+                start_col = item->column();
+            }
+            if (item->column() > end_col)
+            {
+                end_col = item->column();
+            }
+        }
+
+        for (int row=start_row; row <= end_row; ++row)
+        {
+            for (int col=start_col; col <= end_col; ++col)
+            {
+                QTableWidgetItem *item = ui.tableWidget->item(row, col);
+                copied_text += item->text();
+                item->setText("");
+                if (col < ui.tableWidget->columnCount() - 1)
+                {
+                    copied_text += "\t";
+                }
+            }
+            copied_text += "\n";
+        }
+        QClipboard *clipboard = QGuiApplication::clipboard();
+        clipboard->setText(copied_text);
+    }
+
+    // Table Paste
+    // Allows pasting of cell data from spreadsheet applications such as LibreOffice Calc.
+    if ((event->modifiers() == Qt::CTRL) && (event->key() == Qt::Key_V))
+    {
+        QClipboard *clipboard = QGuiApplication::clipboard();
+        QString pasted_text = clipboard->text();
+        QList<QString> rows = pasted_text.split('\n');
```

Comment:
should be changed to
`QList<QString> rows = pasted_text.split('\n',Qt::SkipEmptyParts);`

Otherwise always a new empty line is added after the pasted content.

Reply:
Thanks and good catch.